docs/3.0/selector.md

```diff
@@ -0,0 +1,389 @@
+
+### Star selector
+
+The character `'*'` matches all widgets. Sometimes useful in conjunction with the [`apply`](#apply) method or [Relationship Selectors](#relationship-selectors):
+
+```js
+page.find(':host > * > Button')
+```
+
+This selects all Buttons attached to any child of *page*, regardless of the type of the child in between.
+
+### Selector Functions
+
+Filter functions are also valid selectors. These type of functions are called for each widget candidate and need to return either `true` to include it or `false` to exclude it. For example, the following snippet would select all visible widgets on a page:
+
+```js
+page.children(widget => widget.visible)
+```
+
+When the given function is a constructor it will _not_ be called, but treated as a [type selector](#type-selectors).
+
+## WidgetCollection
+
+Instances of `WidgetCollection` can be both the basis and the result of a selection. Therefore a quick overview is useful here:
+
+Widget collections are array-like objects that represent a set of widgets in Tabris.js. They are immutable and every entry is unique, i.e. they never contain any duplicates. They may also be empty, representing zero widgets.
+
+ Notably, a widget collection features a subset of widget API that allows modifying all contained widgets at once. This includes `set`, `trigger`, `on`, `off`, `once`, `append`, `appendTo`, `dispose` and `animate`. Example:
+
+```js
+collection
+  .set({background: 'blue'}) // make all widgets blue..
+  .animate({opacity: 0}, {duration: 400}); // fade out
+```
+
+The same could be done to only a specific entry:
+
+```js
+collection[3]
+  .set({background: 'blue'})
+  .animate({opacity: 0}, {duration: 400});
+```
+
+The first example will never fail, even if the collection is empty, while the second may cause an exception if the collection has less than four widgets. Like arrays, widget collections are zero indexed.
+
+While `WidgetCollection` has some array-like API (`length`, `forEach`, `indexOf`), you can also get an actual array, which is mutable safe copy:
+
+```js
+const arr = collection.toArray();
+arr.splice(2, 3);
+const collection2 = new WidgetCollection(arr);
+```
+
+JSX can be used to create a widget collection, usually to create and append an entire ui tree:
+
+```jsx
+contentView.append(
+  <WidgetCollection>
+    <TextView />
+    <TextView />
+    <TextView />
+  </WidgetCollection>
+);
+```
+
+In TypeScript `WidgetCollection` is a generic type (`WidgetCollection<T extends Widget = Widget>`>that "knows" what type of widgets are contained - if they are all of the same type:
+
+```js
+const collection: WidgetCollection<TextView> = new WidgetCollection([new TextView()]);
+collection[0].text = 'foo'; // would not compile on WidgetCollection<Widget>
+```
+
+Such a collection is created implicitly whenever a constructor is used as a [type selector](#type-selectors).
+
+## APIs that accept Selectors
+
+### Composite#children
```

Comment:
How common is "#" to reference an instance member? Should we do `composite.children()` instead?